.github/workflows/ci.yml

```diff
@@ -0,0 +1,69 @@
+name: CI
+
+on:
+    push:
+        branches: [ main ]
+    pull_request:
+    schedule:
+        -   cron: '0 0 * * MON'
+
+jobs:
+    check-cs:
+        name: Check Coding Standards
+        runs-on: ubuntu-latest
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: PHP-CS-Fixer
+              uses: docker://oskarstark/php-cs-fixer-ga
+              with:
+                args: --config=.php_cs --diff --dry-run
+
+    ci:
+        name: Test PHP ${{ matrix.php-version }} ${{ matrix.name }}
+        runs-on: ubuntu-latest
+        strategy:
+            fail-fast: false
+            matrix:
+                php-version: ['7.1', '7.2', '7.3', '7.4', '8.0']
+                minimum-stability: ['']
+                composer-flags: ['']
+                name: ['']
+                include:
+                    - php-version: 7.1
+                      composer-flags: '--prefer-lowest'
+                      name: '(prefer lowest dependencies)'
+                    - php-version: 8.0
+                      minimum-stability: 'dev'
+                      name: '(minimum dev stability)'
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Setup PHP, with composer and extensions
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: ${{ matrix.php-version }}
+                  extensions: mbstring, xml
+                  ini-values: phar.readonly="Off"
+
+            - name: Get composer cache directory
+              id: composer-cache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Cache composer dependencies
+              uses: actions/cache@v2
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: ${{ runner.os }}-composer-${{ github.sha }}
```

Comment:
Actually, I would remove the "github.sha" from the key, It scatters the cache, and so is kind of useless.

Reply:
I will fix it later, thanks :+1: